test/functional/aws_test.go

```diff
@@ -0,0 +1,242 @@
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		{
+			DBName:             aws.String(UPSResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		{
+			DBName:             aws.String(CRResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		{
+			DBName:             aws.String(CRResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		//TODO implement Fuse here when it's ready
+		{
+			DBName:             aws.String(ClusterSSOResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+		{
+			DBName:             aws.String(ApplicationSSOResourceName),
+			DeletionProtection: aws.Bool(true),
+			MultiAZ:            aws.Bool(true),
+			StorageEncrypted:   aws.Bool(true),
+		},
+	}
+	expectedElasticacheResources = []*elasticache.ReplicationGroup{
+		{
+			AtRestEncryptionEnabled:  aws.Bool(true),
+			TransitEncryptionEnabled: aws.Bool(true),
+		},
+	}
+	expectedS3Resources = []*s3.PublicAccessBlockConfiguration{
+		{
+			BlockPublicAcls:       aws.Bool(true),
+			BlockPublicPolicy:     aws.Bool(true),
+			IgnorePublicAcls:      aws.Bool(true),
+			RestrictPublicBuckets: aws.Bool(true),
+		},
+	}
+)
+
+func AWSResourcesExistTest(t *testing.T, ctx *TestingContext) {
+	err := setAWSCredEnvar(ctx.framework)
+	if err != nil {
+		t.Errorf("failed to set AWS cred envars : %w", err)
+	}
+	clusterID, err := common.GetClusterID(ctx.framework)
+	if err != nil {
+		t.Errorf("failed to get clusterID : %w", err)
+	}
+	instances, instanceErr := getRDSResources(*newAWSClients())
+	elasticacheResourcesMap, elasticacheErr := getElasticacheResources(*newAWSClients(), clusterID)
+	s3ResourcesMap, s3Err := getS3Resources(*newAWSClients(), clusterID)
+	if instanceErr != nil || elasticacheErr != nil || s3Err != nil {
+		t.Errorf("error getting one or more aws resource")
+	}
+	resourceMatchesExpectedState := make(map[string]bool)
+	//Check RDS
+	for _, expectedInstance := range expectedRDSResources {
+		for _, returnedInstance := range instances {
+			if expectedInstance.DBName != returnedInstance.DBName {
+				continue
+			}
+			if !common.EqualResourceBool(
+				[]bool{aws.BoolValue(expectedInstance.DeletionProtection), aws.BoolValue(expectedInstance.MultiAZ), aws.BoolValue(expectedInstance.StorageEncrypted)},
+				[]bool{aws.BoolValue(returnedInstance.DeletionProtection), aws.BoolValue(returnedInstance.MultiAZ), aws.BoolValue(returnedInstance.StorageEncrypted)}) {
+				t.Logf("RDS resources are not in the expected state")
+				resourceMatchesExpectedState[aws.StringValue(expectedInstance.DBName)] = false
+			}
+			resourceMatchesExpectedState[aws.StringValue(expectedInstance.DBName)] = true
```

Comment:
Would not it be marked as true every time? Should this be in `else` branch?

Reply:
Good catch! completely missed that when I put it in :+1: